src/main/java/bio/terra/cda/app/controller/MetaApiController.java

```diff
@@ -30,16 +36,27 @@ public MetaApiController(ApplicationConfiguration applicationConfiguration) {
 
   @Override
   public ResponseEntity<SystemStatus> serviceStatus() {
-    HttpStatus httpStatus = HttpStatus.OK;
-
-    SystemStatusSystems otherSystemStatus =
-        new SystemStatusSystems().ok(true).addMessagesItem("everything is fine");
+    SystemStatus systemStatus = new SystemStatus();
+    try {
+      BigQuery bigQuery = BigQueryOptions.newBuilder().setProjectId("gdc-bq-sample").build().getService();
+      String StatusCheck = bigQuery.getDataset("cda_mvp").getDatasetId().getDataset();
+      if (StatusCheck.equals("cda_mvp")) {
+        SystemStatusSystems otherSystemStatus = new SystemStatusSystems().ok(true).addMessagesItem("everything is fine");
+        systemStatus.ok(true).putSystemsItem("BigQuery", otherSystemStatus);
+        return new ResponseEntity<>(systemStatus, HttpStatus.OK);
+      }
+    } catch (Exception e) {
+      logger.error(e.getMessage());
+      SystemStatusSystems otherSystemStatus = new SystemStatusSystems();
```

Comment:
There's no need to have two different `SystemStatusSystems` objects. If you create this outside the `try` and call it something like `bigQuerySystemStatus`, you can create it once and update it in both the `try` and `catch` blocks.